main.py

```diff
@@ -24,19 +25,64 @@ class Main(KytosNApp):
     spec = load_spec(pathlib.Path(__file__).parent / "openapi.yml")
 
     def setup(self):
-        """Replace the '__init__' method for the KytosNApp subclass.
-
-        The setup method is automatically called by the controller when your
+        """
+        The setup method is automatically called by the controller when the
         application is loaded.
-
-        So, if you have any setup routine, insert it here.
         """
         self.pipeline_controller = self.get_pipeline_controller()
+        self.required_napps = set()
+        self.current_enabled_table = dict()
+        self.load_enable_table(self.get_enabled_table())
 
     def execute(self):
         """Execute once when the napp is running."""
         log.info("Executing of_multi_table")
 
+    def get_enabled_table(self):
+        """Get the only enabled table, if exists"""
+        return self.pipeline_controller.get_pipelines("enabled")
+
+    def load_enable_table(self, pipeline):
+        """Seach for a table that is enabled in the database
+        If there is initiate enable_table process"""
+        if not pipeline:
+            # There is not pipeline with {"status": "enabled"}
+            return
+
+        found_napps = set()
+        enable_napps = self.get_enabled_napps()
+        # There is always maximum 1 pipeline enabled
+        for table in pipeline["pipelines"][0]["multi_table"]:
+            for napp in table["napps_table_groups"]:
+                if napp in enable_napps:
+                    found_napps.add(napp)
+        self.required_napps = found_napps
+        self.current_enabled_table = pipeline["pipelines"][0]
+        self.start_enabling_pipeline()
+
+    def get_enabled_napps(self):
+        """Get the NApps that are enabled and subscribed"""
+        enable_napps = set()
+        for key in self.controller.napps:
+            # Keys look like this: ('kytos', 'of_lldp')
+            if key[1] in SUBSCRIBED_NAPPS:
+                enable_napps.add(key[1])
+        return enable_napps
+
+    def start_enabling_pipeline(self):
+        """Method to start the process to enable table
+        First, build content to be sent through an event
+        """
+        content = {}
+        for table in self.current_enabled_table["multi_table"]:
```

Comment:
Depending how `start_enabling_pipeline` and `load_enable_table` are called you'd end up with a Thread race conditions since you're sharing `current_enabled_table`, you might need a flag or status with a threading Lock, we probably don't want this being potentially be executed from multiple threads once `POST `/api/kytos/of_multi_table/v1/pipeline/<pipeline_id>/enable` could also enable a pipeline.

Reply:
Got it, added